src/App.tsx

```diff
@@ -185,12 +192,50 @@ class App extends Component {
     this.setState({shareId});
     database.setShareId(shareId);
 
-    const updatedNewContext = await getDataContext(dataContextName);
+    const updatedNewContext = await Codap.getDataContext(dataContextName);
     if (updatedNewContext) {
       database.set("dataContext", updatedNewContext);
     }
   }
 
+  joinShare = async () => {
+    const shareId = this.state.joinShareId;
+    this.setState({shareId});
+    database.setShareId(shareId);
+
+    const response = await database.getAll();
+    const contextData = response && response.val();
+    if (contextData) {
+      const { dataContext, items } = contextData;
+
+      // fix parent references -- assumes collections are written out in order
```

Comment:
Is this a CODAP issue? I see that when we serialize the data context, the collections get parents which are the id of the parent. But CODAP expects them to be the name of the parent?

Is this something that should be fixed on CODAP's side?

Also, instead of assuming that the collections are written out in order, could we find the one with the right id?

Reply:
I believe the parent IDs are ephemeral. I think the right thing to do might be to have a serialization process for DataContexts that converts ephemeral IDs to names where appropriate and could also eliminate some fields that don't need to be serialized, for instance.